### PR TITLE
[Reports] Move print_version_info

### LINKED ISSUE #8623: PTR Flag not working for spell_query
### Describe the bug
Running `simc ptr=1 spell_query="x"` will output the same as ptr=0

### Expected behavior
spell_query should run in ptr mode when ptr is set to 1 

### To Reproduce
Run the spell_query command manually or run the spelldatadump_ptr.bat file in /SpellDataDump, the output will be Live data and the build info will also output live.

### Additional information
* SimulationCraft Version (required)
Latest

* If SimulationCraft is crashing, please include your **dxdiag** file - Instructions here - 
N/A

* **simc_gui.simc** file or contents
N/A

### PR DESCRIPTION
**Pull request by JeremyMonk**
* Move print_version_info to after command arguments are parsed and setup of control
* Fixes #8623

@vituscze

### CODE REVIEW

**Comment by gastank** [comment]
This would result in any options parse failure not giving the version info at all. Version info should always be displayed for all outputs, valid or invalid. If you want to display the PTR string post-parsing the string can be moved to a post-parse function like print_build_info().

File: engine/sc_main.cpp [hunk 1]
```diff
@@ -282,12 +293,14 @@ int sim_t::main( const std::vector<std::string>& args )
...
     {
+      _print_version_info();
       fmt::print( "\n" );
       std::throw_with_nested( std::runtime_error( "Setup failure" ) );
     }
 
-    // print version info
-    print_version_info( *dbc );
+    // print version info if it hasn't been displayed already
+    if ( !version_displayed )
```
Comment (vituscze) [inline engine/sc_main.cpp]: Any reason this check is here twice?

Comment (JeremyMonk) [reply engine/sc_main.cpp]: Forgot to remove it but I think the lambda is unnecessary anyway as code execution stops after throw_with_nested is called

**Comment by vituscze** [comment]
Looks reasonable to me now.

**Pull request merged**